feat: add cms component schematic

znikola opened the pull request:
closes #12

psmul commented on projects/schematics/src/add-cms-component/index.ts:
```diff
@@ -0,0 +1,390 @@
+      const insertion =
+        `  <ng-container *ngIf="${CMS_COMPONENT_DATA_PROPERTY_NAME}$ | async as data">` +
+        `\n      {{data | json}}` +
+        `\n    </ng-container>`;
```
When component is created, result html has weird intendation.

znikola replied:
Is there an API that can return us the existing indentation level?

znikola replied:
Scratch that, I decided to put everything in one line

psmul commented on projects/schematics/src/add-cms-component/schema.json:
```diff
@@ -0,0 +1,160 @@
+  "title": "Spartacus Schematics",
+  "type": "object",
+  "properties": {
+    "declaringCmsModule": {
```
declaringCmsModule sounds/reads weird to me, maybe just `declareCmsModule`?

znikola commented:
>     1. Lack of created cms component import in imports ConfigModule configuration

Good catch, this is caused when `skipImport=true`. 
 
>     2. Maybe OnInit is unnecessary when we create a new component?

That's how Angular generates the component by default, and there's no an option to easily switching it off. Do you think it's worth the effort to remove it after generating?

>     3. When adding created cms component to app.module.ts, a glitch with 2 commas in imports appears

I can't reproduce this with `ng g .:add-cms-component myAwesomeCms --cms-model=MyModel --declareCmsModule=app`. Can you explain it to me how to reproduce it?

znikola commented:
Responding:

> Additional things found after next iteration of QA:
> 
>     1. When used:
>        `:add-cms-component myAwesomeCms --cms-model=MyModel`
>        `my-awesome-cms.module.ts` has wrong component import:
>        `import { MyAwesomeCmsComponent } from './my-awesome-cms.component.ts';`
>        Should not contain '.ts' extension.

Thanks, good catch. Addressed.

>     2. When used:
>        `:add-cms-component myAwesomeCms --cms-model=MyModel`
>        `my-awesome-cms.module.ts` has doubled component import in `imports` (ConfigModule and main imports array)

Again, great catch, addressed.

>     3. I believe we should follow Spartacus module style. If we import component in module file, we should also include declarations like **declarations, entryComponents, exports**. Although if we specify `--module=app` it works fine (adds component to entryComponents and exports).

Good point, implemented.

>     4. When used:
>        `ng g @spartacus/schematics:add-cms-component myAwesomeCms --cms-model=MyModel --module=app --declareCmsModule=app`
>        (I know it's falsy scenario). It also adds module with '.ts' extension.
>        `import { App } from './app.module.ts';`

Addressed with the 1st point.
